aten/src/ATen/native/native_functions.yaml

```diff
@@ -802,12 +802,32 @@
 
 - func: _convolution_double_backward(Tensor? ggI, Tensor? ggW, Tensor? ggb, Tensor gO, Tensor weight, Tensor self, int[] stride, int[] padding, int[] dilation, bool transposed, int[] output_padding, int groups, bool benchmark, bool deterministic, bool cudnn_enabled, bool[3] output_mask) -> (Tensor, Tensor, Tensor)
 
+- func: _convolution_asymmetric_padding_double_backward(Tensor? ggI, Tensor? ggW, Tensor? ggb, Tensor gO, Tensor weight, Tensor self, int[] stride, int[] padding_l, int[] padding_r, int[] dilation, bool transposed, int[] output_padding, int groups, bool benchmark, bool deterministic, bool cudnn_enabled, bool[3] output_mask) -> (Tensor, Tensor, Tensor)
+
 - func: conv1d(Tensor input, Tensor weight, Tensor? bias=None, int[1] stride=1, int[1] padding=0, int[1] dilation=1, int groups=1) -> Tensor
 
 - func: conv2d(Tensor input, Tensor weight, Tensor? bias=None, int[2] stride=1, int[2] padding=0, int[2] dilation=1, int groups=1) -> Tensor
 
 - func: conv3d(Tensor input, Tensor weight, Tensor? bias=None, int[3] stride=1, int[3] padding=0, int[3] dilation=1, int groups=1) -> Tensor
 
+- func: conv1d.padding_modeA(Tensor input, Tensor weight, Tensor bias, int[1] stride, str padding, int[1] dilation=1, int groups=1) -> Tensor
+
+- func: conv1d.padding_modeB(Tensor input, Tensor weight, Tensor bias, *, str padding, int[1] stride=1, int[1] dilation=1, int groups=1) -> Tensor
+
+- func: conv1d.padding_modeC(Tensor input, Tensor weight, *, str padding, Tensor? bias=None, int[1] stride=1, int[1] dilation=1, int groups=1) -> Tensor
```

Comment:
Okay, turns out I was mistaken and `native_functions.yaml` does indeed support `str`. It was actually failing to compile because it won't accept `str padding` as not having a default while the earlier arguments `bias` and `stride` do.

However, if `padding` is given a default value it can be ambiguous with the `int[1]` overload and C++ code fails to compile. At the same time, `bias` and `stride` need to have defaults in python for hen you call it like `conv1d(input, weight, padding='same')`. I found this ugly workaround of defining these two extra overloads where padding is keyword-only and to the left of all the defaulted arguments.

Reply:
ooooooof